File: leetcode/p1701.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        schedule = 0
        total_wait = 0
        for a, t in customers:
            schedule = max(a, schedule) + t
            total_wait += schedule - a
        return total_wait / len(customers)

File: leetcode/p1701.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        int cur = 0;
        long long int total = 0;
        for (int i=0;i<customers.size();i++) {
            cur = (cur>customers[i][0] ? cur : customers[i][0]) + customers[i][1];
            total += cur - customers[i][0];
        }
        return total / double(customers.size());
    }
};
